poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java

```diff
@@ -1086,38 +1215,51 @@ public PowerIQAPIClient createClient(FacilitySoftwareConfig poweriq) {
    }
 
    public List<Asset> filterAssetsBySource(String source, Set<String> assetIds) {
-      List<Asset> assets = new ArrayList<Asset>();
-      List<Asset> assetsFromPowerIQ = restClient.getAllAssetsBySource(source);
-      for (Asset asset : assetsFromPowerIQ) {
-         if (assetIds.contains(asset.getId())) {
-            assets.add(asset);
+      List<Asset> sensorAssets = new ArrayList<Asset>();
```

Comment:
This logic is wired. Why do we want change restClient.getallAssetsBySource ?  it will just use 1 api call to get all the assets. the new change will call a bunch of apis.

Reply:
/source/{assetsource}  we already has this api.  let's query it by 100 per page. then it will reduce the api calls in 100 times.

Reply:
Reverted